spec/android_localize_helper_spec.rb

```diff
@@ -117,12 +117,52 @@ def generated_file(code)
           end
         end
 
+        shared_examples 'en-dash substitutions' do |xpath, fixture_block, gp_block, final_block|
+          it 'has at least one fixture with text to be substituted' do
+            # This ensures that even if we modify the fixtures in the future, we will still have a case which tests this
+            gp_xml = File.open(stub_file('pt-br')) { |f| Nokogiri::XML(f, nil, Encoding::UTF_8.to_s) }
+            gp_node = gp_xml.xpath(xpath).first
+            expect(gp_node.content).to include(fixture_block)
+          end
+
+          TEST_LOCALES_MAP.each do |h|
+            it "has the text substituted in #{h[:android]}" do
+              final_xml = File.open(generated_file(h[:android])) { |f| Nokogiri::XML(f, nil, Encoding::UTF_8.to_s) }
+              final_node = final_xml.xpath(xpath).first
+              expect(final_node.content).to include(final_block)
+              expect(final_node.content).not_to include(gp_block)
+            end
+          end
+        end
+
         context 'with //string tags' do
-          include_examples 'substitutions', "/resources/string[@name='shipping_label_payments_saving_dialog_message']"
+          include_examples 'ellipsis substitutions', "/resources/string[@name='shipping_label_payments_saving_dialog_message']"
+
+          # en-dash: substitute ranges
+          include_examples 'en-dash substitutions', "/resources/string[@name='threat_fix_description']", '0-1', '0-1', '0–1'
+          include_examples 'en-dash substitutions', "/resources/string[@name='threat_fix_description_large']", '0 - 1', '0 - 1', '0 – 1'
+
+          # en-dash: don't substitute negative numbers
+          include_examples 'en-dash substitutions', "/resources/string[@name='field_allowed_values']", '1 -1 -2', '1 –1 –2', '1 -1 -2'
+
+          # en-dash: don't substitute checklists
+          include_examples 'en-dash substitutions', "/resources/string[@name='checklist_one']", '- 1.', '– 1.', '- 1.'
+          include_examples 'en-dash substitutions', "/resources/string[@name='checklist_two']", '- o', '– o', '- o'
         end
 
```

Comment:
There may be a lot more cases which can be tested, but I think these are the most relevant ones.

Reply:
I think it's worth adding cases where:
 - we have multiple ranges in a single string (e.g. ``abc 1-2 def 3-4 ghi $5-$6 jkl'`)
 - we have ranges that are not just numbers, but numbers with currency symbols too (e.g. `$70-$120` – which is the exact case that we stubbled upon in WP – and `70€-120€`)

Reply:
I added a test case for the first. 
I don't think the second works at all with the current regex and since it's the same rule used by the linter, I'd opt to ship this version and iterate in the future if the need will arise.